standardtracer/propagation.go

```diff
@@ -177,52 +181,48 @@ func (p *splitBinaryPropagator) JoinTrace(
 	var traceID, propagatedSpanID int64
 	var sampledByte byte
 
-	err = binary.Read(contextReader, binary.BigEndian, &traceID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &traceID); err != nil {
 		return nil, opentracing.ErrTraceCorrupted
 	}
-	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &propagatedSpanID); err != nil {
```

Comment:
What's the original plan for `standardtracer`? I thought it was going to be moved out of this package into its own (which is fine).

I don't understand your suggestion - surely `standardtracer` would make some assumptions about itself (looks like it's being dapperish), and so its serialization logic would as well). Are you suggesting pulling out some of the encoding logic into a more reusable form for other types of tracers?